aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AutoSessionTracker.java

```diff
@@ -77,20 +75,19 @@ void stopSessionTracking(final Application application) {
     @Override
     public void onActivityCreated(final Activity activity, final Bundle bundle) {
         Log.d(LOG_TAG, "onActivityCreated " + activity.getLocalClassName());
-        handleOnCreateOrOnStartToHandleApplicationEnteredForeground();
         activityLifecycleStateMap.put(activity, Lifecycle.State.CREATED);
     }
 
     @Override
     public void onActivityStarted(final Activity activity) {
         Log.d(LOG_TAG, "onActivityStarted " + activity.getLocalClassName());
-        handleOnCreateOrOnStartToHandleApplicationEnteredForeground();
         activityLifecycleStateMap.put(activity, Lifecycle.State.STARTED);
     }
 
     @Override
     public void onActivityResumed(final Activity activity) {
         Log.d(LOG_TAG, "onActivityResumed " + activity.getLocalClassName());
+        handleOnCreateOrOnStartToHandleApplicationEnteredForeground();
```

Comment:
This method name is long to the point of being unreadable. Maybe just inline that method, here? Or, name it like `checkForApplicationEnteredForeground()`?  As a bonus, this would match your `checkForApplicationBackground()` method.

Reply:
I have some more clean-ups lined up where I have this change, cleaning up comments and removing Screen off receiver because it turn out screen turn off also causes running activity to stop thus giving a onStop callback. This PR is WIP and there is no way to make it a Draft PR once its marked ready for review :/